File: 12integerToRoman/integerToRoman.java
enum Roman
{
    M(1000), CM(900), D(500), CD(400), C(100), XC(90), L(50), XL(40), X(10), IX(9), V(5), IV(4), I(1);
    private final int roman;
    private Roman(final int roman)
    {
        this.roman = roman;
    }
    public int value()
    {
        return this.roman;
    }
}

class Solution {
    public String intToRoman(int num) {
        final var result = new StringBuilder();
        for (final var $: Roman.values())
        {
            result.append($.name().repeat(num / $.value()));
            num %= $.value();
        }
        return result.toString();
    }
}

File: 12integerToRoman/integerToRoman.py
import enum

@enum.unique
class Roman(enum.Enum):
    M = 1000
    CM = 900
    D = 500
    CD = 400
    C = 100
    XC = 90
    L = 50
    XL = 40
    X = 10
    IX = 9
    V = 5
    IV = 4
    I = 1

class Solution:
    def intToRoman(self, num: int) -> str:
        result = []
        for _ in Roman:
            result += itertools.repeat(_.name, num // _.value)
            num %= _.value
        return ''.join(result)
